### DISCUSSION TITLE
Ballpark speed comparison vs alternatives

### DISCUSSION THREAD

**Discussion by NickCrews** [GRAAL-Research/deepparse#233] [Q&A] [answered]
Hey there! Just discovered this software. I've used [libpostal](https://github.com/openvenues/libpostal) and [usaddress](https://github.com/datamade/usaddress) before. The developer experience definitely looks better and more modern here, but I'm wondering if you have any ballpark estimate on how the speed performance varies between these options? I am doing bulk processing (~10 million addresses), so speed is moderately important. Thank you!

**Comment by davebulaval** [comment, marked as answer]
Hi there!

Thank you for your interest in Deepparse. Deepparse was designed to offer a more modern and better development experience than other solutions (e.g. an easy-to-use Docker image and API deployment).

We have empirically analyzed the performance of `libpostal` and Deepparse but not with `usaddress`. The details are available [here](https://deepparse.org/parser.html#memory-usage-and-time-performance). 

If you like to test `usaddress` and add your results to our table, I would greatly appreciate a PR for this. You can find the script used for the speed tests [here](https://github.com/GRAAL-Research/deepparse/tree/main/models_evaluation).

**Reply by NickCrews** [reply]
Thank you! that is just what I was looking for.

**Reply by davebulaval** [reply]
Do you plant to open a PR with `usaddress` results?

**Reply by NickCrews** [reply]
unfortunately, I don't think I will pursue this library, libpostal appears to be so much faster that I think I will have to stick with it's worse ergonomics. If you manage to get the performance more comparable, I would love to revisit deepparse! Really appreciate your work.

**Reply by davebulaval** [reply]
It is normal; Libpostal is written in C. But, if you do not need online processing, it should not bother you. 

Transferring the library to Rust would be the only way to accelerate our processing. It is part of our plan, but not in the short term.

**Comment by aliraza-noon** [comment]
@davebulaval how about the llm models like bert how does it compare ?

**Reply by davebulaval** [reply]
It is a BERT-like Seq2Seq architecture. 
We do not compare to LLM, our architecture is a scaling factor smaller to LLM that such comparison is irrelevant in terms of inference for such specialize task.,
